internal/elasticsearch/common_test.go

```diff
@@ -564,6 +565,36 @@ func TestPodSpecHasTaintTolerations(t *testing.T) {
 	}
 }
 
+func TestElasticSearchSecurityContext(t *testing.T) {
+	podTemplate := newPodTemplateSpec(context.Background(), log.NewLogger("common-testing"), "test-node-name", "test-cluster-name", "test-namespace-name", api.ElasticsearchNode{}, api.ElasticsearchNodeSpec{}, map[string]string{}, map[api.ElasticsearchNodeRole]bool{}, nil, LogConfig{})
+
+	if !pointer.BoolEqual(podTemplate.Spec.SecurityContext.RunAsNonRoot, pointer.Bool(true)) {
```

Comment:
IMHO this should used differently:
```go
if pointer.BoolEqual(false, podTemplate.Spec.SecurityContext.RunAsNonRoot)
```
AFAICS the second parameter is expected to be a `*bool`

Reply:
they are both expected to be a pointer: `func pointer.BoolEqual(a *bool, b *bool) bool`